src/conv-core/cpuaffinity.C

```diff
@@ -442,62 +448,149 @@ static void cpuAffinityHandler(void *m)
 */
   CmiAssert(CmiMyPe()==0 && rankmsg != NULL);
   skt_ip_t & ip = msg->ip;
-  pe = msg->pe;
+  int node = msg->node;
+  int pe = msg->pe;
   auto iter = hostTable.find(ip);
   if (iter != hostTable.end()) {
     rec = iter->second;
     CmiFree(msg);
   }
   else {
     rec = msg;
-    rec->seq = nodecount;
-    nodecount++;                          /* a new node record */
+    rec->seq = hostcount++;
     hostTable.emplace(ip, msg);
   }
-  myrank = rec->rank%rec->ncores;
-  while (in_exclude(myrank)) {             /* skip excluded core */
-    myrank = (myrank+1)%rec->ncores;
-    rec->rank ++;
+
+  int & node_host = rankmsg->node_hosts[node];
+  if (node_host == -1)
+  {
+    rankmsg->node_ranks[node] = rec->node_rank++;
```

Comment:
Shouldn't `rec->node_rank` also be incremented each time a different PE of the same logical node is processed? With the current code, it is only incremented when the first PE of each logical node is processed.

Reply:
No, the variable refers to the rank of the node on the host. I've updated its name to `node_ranks_on_host` to make this clear.